### ISSUE TITLE
Support for Android 4.4 in Emerging Markets 

### ISSUE THREAD

**Issue by lfrodrigues** [react-native-community/discussions-and-proposals#304]
### Introduction

Following the comment I added in react-native-community/releases#214.

The community is planning to remove support for the Android API < 21 in the v0.64 release.
Removing API 19 (Android 4.4) will have a large impact on developers implementing RN apps in Emerging Markets as these countries don't upgrade the phones as often as in Europe/US.

### Details

We develop apps for emerging markets. In some countries in Africa, we still have thousands of users in Android 4.4 (around 4% of the user base in some apps). If you remove support for older SDK, we and everyone else developing for emerging markets won't be able to upgrade and will be stuck in 0.63 for the foreseeable future.

For reference, here's the penetration for Africa:
https://gs.statcounter.com/android-version-market-share/mobile-tablet/africa

### Discussion points

I propose RN keeps support for API 19 (Android 4.4), I think it's a good balance between supporting devices in emerging markets and removing old APIs

**Comment by kuasha420** [comment]
Strong No from me. Time to leave Kitkat behind. Even Lollipop was released 6.5 years ago.

**Comment by mikehardy** [comment]
Fellow emerging markets developer, I also think it's time to move to Android 5+. So it's important to note that you can't say in general "emerging market developers want to stay on Android 4.x". 

Market penetration of 4.x only *appears* to be around 0.7% for Lat Am, but I dug more deeply into the stats on one of my large apps, and I discovered that quite a few of these installs are what I'd call phantoms - the app is still installed but there is a funny property with regarding to the "liveness" of the installs - a lot of them are not being used and I would assume the phone has been passed down to a child in the house or is sitting in a drawer or something so it gets an update but I see no real active sessions from them.

More important point: this whole argument is after the fact. It has already happened. It's merged and released now. So we are just wasting our time with every single character typed here.

The reason I'm replying then is to say you should branch your app and keep the min SDK you need for your app on the branch, and don't upgrade the RN version in that branch. Cherry pick whatever features and bug fixes you need on to the branch and keep doing releases for the archaic devices with lower version codes (but higher than previous releases for the archaic devices

On your main branch upgrade react-native and bump the minSdk and put out releases for the modern devices.

In other words, it's a release engineering / version management issue, and no reason to hold back the software ecosystems of the world. Just getting rid of multidex is such a huge win...it's time

**Comment by lfrodrigues** [comment]
I guess I should have written in Africa. I assumed the problem existed everywhere and it doesn't. 

We get a few 1000s of new installs per day that are 4.4. In the older versions, you're right, we don't have active users.

If you look at the trend it's doing down in Africa but the reality is that there a lot more people using 4.4 devices than anywhere in the world. I see the trend in our data that align with the link I shared above. 

I understand everyone wants to remove older versions of APIs but the reality is that the different regions don't move at the same speed. At least for us, and I assume many more people with a large install base in Africa, would be better if you keep android 4.4 for one additional release and remove it in the next.

**Comment by kuasha420** [comment]
Was the dropping of API Level Below 21 discussed publicly before the patchset got merged or was it an internal decision? From the commit message it looks like the initial plan was to support API Level 19 but then suddenly went to 21.

**Comment by newyankeecodeshop** [comment]
It might have had something to do with security on the platform. The docs for OkHttp say:
> The OkHttp 3.12.x branch supports Android 2.3+ (API level 9+) and Java 7+. These platforms lack support for TLS 1.2 and should not be used. 
https://github.com/square/okhttp#requirements

Perhaps the lack of TLS 1.2 on 4.4 played a role?

**Comment by TheSavior** [comment]
Because React Native exists in our monorepo at FB, we take on the same sdk version requirements as the rest of Facebook's products. In general, because FB products have something like 3B people using them around the world, that is intended to give the ecosystem good coverage of devices and countries. FB has dropped support for Android API < 21, and thus React Native has as well. If we tried to diverge from the rest of the FB ecosystem and had further back support, we would have no coverage of that because no users would be on it. React Native would likely break much more frequently for you on those versions, which would probably be worse than dropping it in a new release of React Native. 😕

**Comment by halaei** [comment]
I think it is possible to enable TLSv1.2 on android 4. I don't remember what we exactly did for that.
We also have 1.9% users on Android 4 in our app. After updating to RN 0.64, I probably keep releasing a version for Android 4 with RN 0.63 for a while, unless it gets too complicated.
